mbed_greentea/mbed_greentea_cli.py

```diff
@@ -616,29 +655,10 @@ def main_cli(opts, args, gt_instance_uuid=None):
                         binary_type=binary_type)
                     #TODO no tests to execute
 
-                filtered_ctest_test_list = ctest_test_list
-                test_list = None
-                if opts.test_by_names:
-                    filtered_ctest_test_list = {}   # Subset of 'ctest_test_list'
-                    test_list = opts.test_by_names.split(',')
-                    gt_logger.gt_log("test case filter (specified with -n option)")
-
-                    invalid_test_names = False
-                    for test_name in test_list:
-                        if test_name not in ctest_test_list:
-                            invalid_test_names = True
-                        else:
-                            gt_logger.gt_log_tab("test filtered in '%s'"% gt_logger.gt_bright(test_name))
-                            filtered_ctest_test_list[test_name] = ctest_test_list[test_name]
-                    if invalid_test_names:
-                        gt_logger.gt_log_warn("invalid test case names (specified with -n option)")
-                        for test_name in test_list:
-                            if test_name not in ctest_test_list:
-                                gt_logger.gt_log_warn("test name '%s' not found in CTestTestFile.cmake (specified with -n option)"% gt_logger.gt_bright(test_name))
-                        gt_logger.gt_log_tab("note: test case names are case sensitive")
-                        gt_logger.gt_log_tab("note: see list of available test cases below")
-                        list_binaries_for_targets(verbose_footer=False)
-
+                filtered_ctest_test_list = dict(ctest_test_list)
```

Comment:
I think this whole block
```python
filtered_ctest_test_list = dict(ctest_test_list)
if opts.test_by_names or opts.skip_test:
    filtered_ctest_test_list = create_filtered_test_list(filtered_ctest_test_list, opts)
```
can be replaced with:
```python
filtered_ctest_test_list = create_filtered_test_list(filtered_ctest_test_list, test_by_names, skip_test)
```

I think your function should just do nothing if ```test_by_names``` & ```skip_test``` are ```None```